### PR TITLE
Restrictions

### PR DESCRIPTION
**Pull request by kathytbui**
### New Feature Submissions:
* [x] Have you added an explanation of what your changes do?
* [x] Have you linted your code locally prior to submission?

### Testing:
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?
* [x] All new and existing tests passed?

### Test coverage:
simplecov - 226/226 lines covered - testing for search controller and user controller needs to be revisited
rubocop - no offenses
test suite - 100%

--------------------------------------------------------------------------------
### Change Log:
- Create spec to test for visitor restrcitions on site 
- Expect visitors to get a 404 response when trying to visit /search or /dashboard
- Add visitor restrictions to all controllers

### CODE REVIEW

File: app/controllers/search_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class SearchController < ApplicationController
+  before_action :require_user
   def index
-    conn = Faraday.get("http://localhost:9292/images/#{params[:search]}")
+    conn = Faraday.get("https://damp-forest-93176.herokuapp.com/images/#{params[:search]}")
```
Comment (janegreene) [inline app/controllers/search_controller.rb:4]: "damp forest" live in action!

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 class ImagesController < ApplicationController
+  before_action :require_user
```
Comment (sciencefixion) [inline app/controllers/images_controller.rb:2]: I like the naming on this method. It's very descriptive.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 class UsersController < ApplicationController
+  before_action :require_user
```
Comment (janegreene) [inline app/controllers/users_controller.rb:2]: I dig this simple guard action!

File: app/controllers/search_controller.rb (hunk 1 above)
Comment (sciencefixion) [inline app/controllers/search_controller.rb:4]: These names though... XD

File: spec/features/restrictions/visitor_page_restrictions_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  describe 'As a visitor' do
+    it "renders 404 when visiting any page other than welcome page" do
+      visit '/dashboard'
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+
+      visit '/search'
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+    end
+  end
+end
```
Comment (sciencefixion) [inline spec/features/restrictions/visitor_page_restrictions_spec.rb:13]: Great spec!

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (dtsehl) [inline app/controllers/users_controller.rb:2]: This is a good piece of functionality but I wonder if there's a way to not need this in every controller?

**Review by janegreene — APPROVED** [review]
Looks great, nice job!

**Review by sciencefixion — APPROVED** [review]
Excellent work! You just knocked out a big blocker for us with this!

**Pull request merged**